renku/api/datasets.py

```diff
@@ -106,37 +106,37 @@ def store_dataset(self, dataset):
     def with_dataset(self, name=None):
         """Yield an editable metadata object for a dataset."""
         from renku.models.refs import LinkReference
-        with self.lock:
-            dataset = self.load_dataset(name=name)
 
-            if dataset is None:
-                dataset = Dataset(name=name)
-                setattr(dataset, '__source__', {})
+        dataset = self.load_dataset(name=name)
 
-                path = (
-                    self.renku_datasets_path / dataset.identifier.hex /
-                    self.METADATA
-                )
-                path.parent.mkdir(parents=True, exist_ok=True)
-                setattr(dataset, '__reference__', path)
+        if dataset is None:
+            dataset = Dataset(name=name)
+            setattr(dataset, '__source__', {})
 
-                if name:
-                    LinkReference.create(
-                        client=self, name='datasets/' + name
-                    ).set_reference(path)
+            path = (
```

Comment:
I think we can update this with dataset_path()

Reply:
will do in another PR

Reply:
well we can't really use the `dataset_path()` here since we don't want to resolve the `LinkReference` in this case.